File: src/dataset/cpp/space-age_exercism-pawelo091991/gemini/GeminiFlash_space_age_v3.cpp
#include "space_age.h"

namespace space_age {

space_age::space_age(double seconds) : time(seconds) {}

double space_age::seconds() const {
  return time;
}

double space_age::on_earth() const {
  return time / earth;
}

double space_age::on_mercury() const {
  return time / mercury_orbital_period;
}

double space_age::on_venus() const {
  return time / venus_orbital_period;
}

double space_age::on_mars() const {
  return time / mars_orbital_period;
}

double space_age::on_jupiter() const {
  return time / jupiter_orbital_period;
}

double space_age::on_saturn() const {
  return time / saturn_orbital_period;
}

double space_age::on_uranus() const {
  return time / uranus_orbital_period;
}

double space_age::on_neptune() const {
  return time / neptune_orbital_period;
}

} // namespace space_age

File: src/dataset/javascript/space-age_exercism-elenamountz/gemini/GeminiFlash_space-age_v3.js
export class SpaceAge {
  constructor(seconds) {
    this.seconds = seconds;
    this.earthYears = seconds / 31557600;
  }

  calculateEarth() {
    return this.earthYears;
  }

  onEarth() {
    return this.roundToTwo(this.earthYears);
  }

  onMercury() {
    return this.roundToTwo(this.earthYears / 0.2408467);
  }

  onVenus() {
    return this.roundToTwo(this.earthYears / 0.61519726);
  }

  onMars() {
    return this.roundToTwo(this.earthYears / 1.8808158);
  }

  onJupiter() {
    return this.roundToTwo(this.earthYears / 11.862615);
  }

  onSaturn() {
    return this.roundToTwo(this.earthYears / 29.447498);
  }

  onUranus() {
    return this.roundToTwo(this.earthYears / 84.016846);
  }

  onNeptune() {
    return this.roundToTwo(this.earthYears / 164.79132);
  }

  roundToTwo(num) {
    return Math.round(num * 100) / 100;
  }
}